dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -89,6 +98,42 @@ async def snapshot(self, req):
             success=True, message="hello", snapshot=snapshot
         )
 
+    @routes.get("/api/component_activities")
```

Comment:
I'm a little concerned about introducing a new dashboard endpoint like this. Are we prepared to handle backwards compatibility for non-existent endpoints? (so far we've mostly just been dealing with new fields). Do we need to be more careful about managing the schema here? (unfortunately the dataclass can't really serve as a model because `rest_response` mutates the fields we return :(

Reply:
Yes, we are only expecting this endpoint to exist from a certain ray version (probably 1.13.2) and will check the ray version before attempting to use.

I agree about the response schema, but do we have a better pattern of specifying this? Currently the unit test of this endpoint is the only thing that validates the response type.